tokio/src/sync/semaphore.rs

```diff
@@ -210,6 +210,55 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// ## Ensure Tests with Mutual Dependencies Run Sequentially
+///
+/// By default, Rust runs tests concurrently. However, in some scenarios, concurrent execution might lead to test interference.
+/// For example, multiple tests might communicate with the same database, necessitating sequential execution to prevent data conflicts.
+///
+/// Consider the following scenario:
+/// 1. `test_insert`: Inserts a key-value pair into the database, then retrieves the value using the same key to verify the insertion.
+/// 2. `test_update`: After insertion, this test updates the key's value to a new one and verifies if the value has been accurately updated.
+/// 3. `test_others`: This is a non-interfering test that can run concurrently with other tests.
+///
+/// For this example, it's essential that `test_insert` and `test_update` run sequentially, though their execution order is inconsequential.
+/// Leveraging a semaphore with a single permit elegantly addresses this challenge.
+///
+/// ```
+/// #[cfg(test)]
+/// mod tests {
+///    use tokio::sync::Semaphore;
+///    // only one permit can be acquired
+///    static PERMIT: Semaphore = Semaphore::const_new(1);
+///    // initilization of database
+///    static DB: database = database::setup();
+///    #[tokio::test]
+///    async fn test_insert() {
+///        // acquire permit first
+///         let permit = PERMIT.acquire().await.unwrap();
+///         let (key, value) = ("name", 0);
+///         DB.insert((key, value)).await;
+///         assert_eq!(DB.get(key), value);
+///         // undo the modification
+///         DB.delete(key).await;
+///         // permit dropped here
+///     }
+///     #[tokio::test]
+///     async fn test_update() {
+///         let permit = PERMIT.acquire().await.unwrap();
+///         let (key, value) = ("name", 0);
+///         DB.insert((key, value)).await;
+///         let new_value = 1;
+///         // update new value at the same key
+///         DB.update((key, new_value)).await;
+///         assert_eq!(DB.get(key).await, new_value);
+///         DB.delete(key).await;
+///    }
+///     #[tokio::test]
+///     async fn test_other() {
+///         // this test can simualtenously run with test_insert and test_update.
+///     }
+/// }
+/// ```
```

Comment:
Please add some empty lines to make this example easier to read.

Reply:
done.